invoices/models.py

```diff
@@ -187,6 +188,106 @@ class Meta:
         ordering = ("-year", "-month", "client", "project")
 
 
+class WeeklyReport(models.Model):
+    WEEKLY_REPORT_STATE_CHOIOCES = (
+        ("C", "Created"),
+        ("A", "Approved")
+    )
+
+    ISSUE_FIELDS = ("billable_incorrect_price_count", "non_billable_hours_count", "non_phase_specific_count", "not_approved_hours_count", "empty_descriptions_count")
+    weekly_report_id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    year = models.IntegerField()
+    week = models.IntegerField()
+
+    project_m = models.ForeignKey("Project", null=True)
+
+    client = models.CharField(max_length=100)
+    project = models.CharField(max_length=100)
+    tags = models.CharField(max_length=1024, null=True, blank=True)
+
+    is_approved = models.BooleanField(blank=True, default=False)
+    has_comments = models.BooleanField(blank=True, default=False)
+    incorrect_entries_count = models.IntegerField(default=0)
+    billable_incorrect_price_count = models.IntegerField(default=0)
+    non_billable_hours_count = models.IntegerField(default=0)
+    non_phase_specific_count = models.IntegerField(default=0)
+    not_approved_hours_count = models.IntegerField(default=0)
+    no_category_count = models.IntegerField(default=0)
+    empty_descriptions_count = models.IntegerField(default=0)
+    bill_rate_avg = models.FloatField(default=0)
+    incurred_hours = models.FloatField(default=0, verbose_name="Incurred hours")
+    incurred_billable_hours = models.FloatField(default=0)
+    billable_percentage = models.FloatField(default=0)
+    incurred_money = models.FloatField(default=0, verbose_name="Incurred money")
+    weekly_report_state = models.CharField(max_length=1, choices=WEEKLY_REPORT_STATE_CHOIOCES, default='C')
+
+    @property
+    def week_start_date(self):
+        return date_utils.week_start_date(self.year, self.week)
+
+    @property
+    def week_end_date(self):
+        return date_utils.week_end_date(self.year, self.week)
+
+    @property
+    def processed_tags(self):
+        if self.tags:
+            return self.tags.split(",")
+        return []
+
+    @property
+    def date(self):
+        return "%s-%02d" % (self.year, self.week)
+
+    @property
+    def full_name(self):
+        return "%s - %s" % (self.client, self.project)
+
+    def __unicode__(self):
+        return u"%s - %s" % (self.full_name, self.date)
+
+    def __str__(self):
+        return u"%s - %s" % (self.full_name, self.date)
+
+    def update_state(self, comment):
+        self.weekly_report_state = "C"
+        if comment.checked:
+            self.weekly_report_state = "A"
+        return self.weekly_report_state
+
+    def get_tags(self):
+        if self.tags:
+            return self.tags.split(",")
+
+    def compare(self, other):
```

Comment:
Is this whole `compare` exact duplicate of what is in `Invoice`? Is it going to stay that for foreseeable future? If the answer to both is yes, you know what needs to be done.